### ISSUE TITLE
About the model parameter

### ISSUE THREAD

**Issue by HaiyongJiang** [vietdoan/Enet_Pytorch#1]
Hi @vietdoan , thank you for sharing your implementation of ENet. Currently, I am trying to train the model  with CamVid dataset. The class accuracy averages to 0.501, while the class iou averages to 0.321. But the result I obtained is far worse than the authors'.  I am not sure if it is possible to share your model parameters? Thank you.

**Comment by vietdoan** [comment]
Hi @HaiyongJiang, I can't reproduce the published result on the paper. The best I got are 0.57 mean acc and 0.44 mean IoU so my model parameters might not be helpful to you. Currently, I am not sure what the problem is.
Please tell me if you improve the performance or find something wrong with the code. Thanks.

**Comment by HaiyongJiang** [comment]
Hi @vietdoan , I found the main problem is that you should use optimizer.zero_grad() rather than optimizer.zero_grad in train.py. After this modification, I can easily reproduce the result. Thank you.

**Comment by vietdoan** [comment]
Thanks a lot for your help! @HaiyongJiang

**Comment by gjy3035** [comment]
@HaiyongJiang I want to know your final performnace. Accoding to my experiments, my encoder can be trained easily. However, after that, the performance of th entire Enet is worse than encoder.

**Comment by HaiyongJiang** [comment]
Hi. I remembered the performance was slightly better than the reported results in the paper with the same parameter configuration. The entire Enet may perform better than encoder. This is because that the outputs of encoder and Enet are different. Encoder holds a smaller segmentation map which may be easier to predict.
